src/helpers/cards.ts

```diff
@@ -321,6 +323,165 @@ export const changeCardStatus = async (
+      client_wallet_account_id: walletId,
+      wallet_type: "GOOGLE",
+      event_type: "TOKEN_STATUS_UPDATED",
+      message_reason: "TOKEN_DEACTIVATED",
+    };
+    await triggerWebhook(CardWebhookEvent.CARD_TOKEN_LIFECYCLE, data);
+  }
+
+  /**
+   * Now go through all creation lifecycle:
+   * - TOKEN_CREATED (INACTIVE)
+   * - LUK_REPLENISHMENT (INACTIVE)
+   * - DEVICE_PROVISIONING_RESULT (INACTIVE)
+   * - OTP_VERIFICATION_RESULT (ACTIVE)
+   */
+  let baseData: ProvisioningTokenStatusChangePayload = {
+    card_id: cardData.card.id,
+    token_reference_id: uuid.v4(),
+    client_wallet_account_id: walletId || uuid.v4(),
+    wallet_type: "GOOGLE",
+  };
+
+  // Token Creation
+  let payload: ProvisioningTokenStatusChangePayload = {
+    ...baseData,
+    token_status: ProvisioningTokenStatus.INACTIVE,
+    event_type: "TOKEN_CREATED",
+    message_reason: "TOKEN_CREATED",
+  }
+  await triggerWebhook(CardWebhookEvent.CARD_TOKEN_LIFECYCLE, payload);
+
+  // From here on event_type is always the same.
+  baseData = {
+    ...baseData,
+    event_type: "TOKEN_STATUS_UPDATED",
+  };
+
+  // Luk replenishment
+  payload = {
+    ...baseData,
+    token_status: ProvisioningTokenStatus.INACTIVE,
+    message_reason: "LUK_REPLENISHMENT",
+  }
+  await triggerWebhook(CardWebhookEvent.CARD_TOKEN_LIFECYCLE, payload);
+
+  // Provisioning Result
+  payload = {
+    ...baseData,
+    token_status: ProvisioningTokenStatus.INACTIVE,
+    message_reason: "DEVICE_PROVISIONING_RESULT",
+  }
+  await triggerWebhook(CardWebhookEvent.CARD_TOKEN_LIFECYCLE, payload);
+
+  // OTP Verification Result
+  payload = {
+    ...baseData,
+    token_status: ProvisioningTokenStatus.ACTIVE,
+    message_reason: "OTP_VERIFICATION_RESULT",
+  }
+  await triggerWebhook(CardWebhookEvent.CARD_TOKEN_LIFECYCLE, payload);
+
+  // Extract unnecessary data to save the token's relevant information .
+  const {
+    event_type,
+    message_reason,
+    wallet_type,
+    ...newProvisioningToken
+  } = payload;
+
+  cardData.provisioningToken = newProvisioningToken;
+  await db.savePerson(person);
+  return newProvisioningToken;
+};
+
+/**
+ * Triggers the CARD_TOKEN_LIFECYCLE webhook with the provided status given an existing person and card.
+ * @param personId {string}
+ * @param cardId {string}
+ * @param status {ProvisioningTokenStatus}
+ */
+export const changeProvisioningTokenStatus = async (
+  personId: string,
+  cardId: string,
+  status: ProvisioningTokenStatus
+): Promise<ProvisioningTokenStatusChangePayload> => {
+
+  if (!personId) {
+    throw new Error("You have to provide personId");
+  }
+  if (!cardId) {
+    throw new Error("You have to provide cardId");
+  }
+
+  const person = await db.getPerson(personId);
+  const cardData = person.account.cards.find(({ card }) => card.id === cardId);
+  if (!cardData) {
+    throw new Error("Card not found");
+  }
+
+  const { provisioningToken } = cardData;
```

Comment:
up to this point in his function, it's the same logic as in `createProvisioningToken`, maybe we could extract and reuse it?

Reply:
b0d94e4